### PR TITLE
compiler and result change for non-agg query

### PR DESCRIPTION
**Pull request by shz117**

### CODE REVIEW

File: query/aql_compiler.go [hunk 1]
```diff
@@ -1775,7 +1784,9 @@ func (qc *AQLQueryContext) processMeasureAndDimensions() {
 			"unsupported aggregate function: %s", aggregate.Name)
 		return
 	}
+}
 
+func (qc *AQLQueryContext) processDimensions() {
```
Comment (voyager-dw) [inline query/aql_compiler.go]: how about no dimension or dimension is "*"?

File: query/common/non_aggregate_query_result.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const MatrixDataKey = "matrixData"
+const HeadersKey = "headers"
+
+// AQLNonAggregatedQueryResult represents non-aggregated query result
+// - there will be a "headers" key, value will be a list of column names
+// - there will be a "matrixData" key, value will be a 2d arary of values (row formated)
+type AQLNonAggregatedQueryResult map[string]interface{}
+
+// Append appends single value to specific row, result rows must be built in order
+func (r AQLNonAggregatedQueryResult) Append(row int, value interface{}) error {
```
Comment (voyager-dw) [inline query/common/non_aggregate_query_result.go]: where to use this new type to return to caller? how is the original aqltimeseriesresult?

Comment (shz117) [reply query/common/non_aggregate_query_result.go]: supposedly in PostProcess and handler. for non-agg query use AQLNonAggregatedQueryResult other wise use time-series-result. we might need to use `map[string]interface{}` in those places and cast these 2 types to map.
another option is to squash everything in the same general result object. wdyt?

File: query/aql_compiler.go [hunk 3]
```diff
@@ -1786,12 +1786,28 @@ func (qc *AQLQueryContext) processMeasure() {
...
+		columns = append(columns, rewrittenVarRef)
+	}
+	return
+}
+
 func (qc *AQLQueryContext) processDimensions() {
 	// Copy dimension ASTs.
-	qc.OOPK.Dimensions = make([]expr.Expr, len(qc.Query.Dimensions))
-	for i, dim := range qc.Query.Dimensions {
+	qc.OOPK.Dimensions = []expr.Expr{}
```
Comment (voyager-dw) [inline query/aql_compiler.go]: how about after we detect it is a non-aggregation query and without dimensions set, then we just add columns into dimension list and then let the compiler automatically handle all other stuffs.

File: query/common/time_series_result.go [hunk 4]
```diff
@@ -73,3 +90,36 @@ func (r AQLTimeSeriesResult) SetHLL(dimValues []*string, hll HLL) {
 		}
 	}
 }
+
+// =====  Time series result methods end =====
+
+// =====  Non aggregate query result methods start =====
+
+// Append appends single value to specific row, result rows must be built in order
+func (r AQLQueryResult) Append(row int, value interface{}) error {
```
Comment (voyager-dw) [inline query/common/time_series_result.go]: likely we only need the interface to add row by row

Comment (shz117) [reply query/common/time_series_result.go]: changed

File: query/aql_compiler.go (hunk 3 above)
Comment (shz117) [reply query/aql_compiler.go]: I changed the implementation to expand wildcard at qc.Query.Dimension level. so processDimensions will remain unchanged.

File: query/common/non_aggregate_query_result.go (hunk 2 above)
Comment (shz117) [reply query/common/non_aggregate_query_result.go]: merged the methods to 1 struct and renamed to `AQLQueryResult`

**Pull request merged**